traversal/procedure/GraphProcedure.java

```diff
@@ -221,6 +230,26 @@ private void registerVertex(PlannerVertex<?> plannerVertex, Set<PlannerVertex<?>
             adjacents.forEach(v -> registerVertex(v, registeredVertices, registeredEdges));
         }
 
+        private void registerEdges(StructureVertex<?> structureVertex) {
+            assert vertices.containsKey(structureVertex.id());
+            structureVertex.outs().forEach(e -> registerEdge(e, true));
+            structureVertex.ins().forEach(e -> registerEdge(e, false));
+            structureVertex.loops().forEach(e -> registerEdge(e, true));
+        }
+
+        private void registerEdge(StructureEdge<?, ?> structureEdge, boolean isForward) {
```

Comment:
builders to create `Procedure` objects from `Structure` are easy because we had part of the code pre-existing!